src/runtime/runtime.cs

```diff
@@ -959,6 +959,58 @@ internal static long PyObject_Size(IntPtr pointer)
         [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
         internal static extern IntPtr PyObject_Dir(IntPtr pointer);
 
+        //====================================================================
+        // Python buffer API
+        //====================================================================
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int PyObject_CheckBuffer(IntPtr obj);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int PyObject_GetBuffer(IntPtr exporter, IntPtr view, int flags);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern void PyBuffer_Release(IntPtr view);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Ansi)]
+        internal static extern long PyBuffer_SizeFromFormat([MarshalAs(UnmanagedType.LPStr)] string format);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int PyBuffer_IsContiguous(IntPtr view, char order);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern IntPtr PyBuffer_GetPointer(IntPtr view, long[] indices);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int PyBuffer_FromContiguous(IntPtr view, IntPtr buf, long len, char fort);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int PyBuffer_ToContiguous(IntPtr buf, IntPtr src, long len, char order);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern void PyBuffer_FillContiguousStrides(int ndims, IntPtr shape, IntPtr strides, int itemsize, char order);
+
+        [DllImport(_PythonDll, CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int PyBuffer_FillInfo(IntPtr view, IntPtr exporter, IntPtr buf, long len, int _readonly, int flags);
+
+        /* buffer interface */
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+        internal struct Py_buffer
+        {
+            public IntPtr buf;
+            public IntPtr obj;        /* owned reference */
+            public long len;
+            public long itemsize;  /* This is Py_ssize_t so it can be
```

Comment:
`Py_size_t` is not the same as `Int64` aka `long`. There is an ongoing discussion about sizes of Python types.

Reply:
Well it works with `long` and i didnt see a reason to use `Py_ssize_t` if it will get casted to long anyway. I looked it up in the python source and it litteraly `typdef`s `Py_ssize_t` as long. What do you recommend to do?

Reply:
@lostmsu What about something like this:    

    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
    internal struct Py_ssize_t
    {
        [MarshalAs(UnmanagedType.SysInt)]
        public IntPtr size;

        public long GetValue()
        {
            if (Is32Bit)
                return size.ToInt32();
            else
                return size.ToInt64();
        }
    }

Reply:
@SnGmng this struct might not work either, because on 64-bit windows `IntPtr` is 64-bit, but `Py_ssize_t` is probably 32-bit. https://stackoverflow.com/questions/384502/what-is-the-bit-size-of-long-on-64-bit-windows

And that problem will have to be solved, before PR is accepted.

Reply:
long and thus Py_ssize_t is 4 bytes on Windows (32 and 64 bit) and 8 bytes on other 64 bit systems. Very annoying.

Reply:
See my comment below, I made this mistake (assuming `Py_ssize_t == long`) before ;)

`Py_ssize_t` is a typedef for `intptr_t` or `ssize_t` (on POSIX), so in all relevant cases it is equivalent in size to `IntPtr`.